game/scripts/vscripts/items/lucience.lua

```diff
@@ -0,0 +1,262 @@
+      unit:RemoveModifierByName("modifier_item_lucience_regen_effect")
+    end
+    -- Force reapplication of effect modifiers so that effects update immediately when Lucience is upgraded
+    foreach(RemoveLucienceEffects, units)
+  end
+end
+
+function modifier_item_lucience_regen_aura:IsHidden()
+  return true
+end
+
+function modifier_item_lucience_regen_aura:IsDebuff()
+  return false
+end
+
+function modifier_item_lucience_regen_aura:IsPurgable()
+  return false
+end
+
+function modifier_item_lucience_regen_aura:IsAura()
+  return true
+end
+
+function modifier_item_lucience_regen_aura:GetAuraRadius()
+  return self:GetAbility():GetSpecialValueFor("aura_radius")
+end
+
+function modifier_item_lucience_regen_aura:GetAuraSearchTeam()
+  return DOTA_UNIT_TARGET_TEAM_FRIENDLY
+end
+
+function modifier_item_lucience_regen_aura:GetAuraSearchType()
+  return DOTA_UNIT_TARGET_HERO + DOTA_UNIT_TARGET_BASIC
+end
+
+function modifier_item_lucience_regen_aura:GetModifierAura()
+  return "modifier_item_lucience_regen_effect"
+end
+
+------------------------------------------------------------------------
+
+modifier_item_lucience_movespeed_aura = class(modifier_item_lucience_regen_aura)
+
+function modifier_item_lucience_movespeed_aura:OnCreated()
+  if IsServer() then
+    local parent = self:GetParent()
+    local units = FindUnitsInRadius(parent:GetTeamNumber(), parent:GetOrigin(), nil, self:GetAbility():GetSpecialValueFor("aura_radius"), self:GetAuraSearchTeam(), self:GetAuraSearchType(), self:GetAuraSearchFlags(), FIND_ANY_ORDER, false)
+
+    function RemoveLucienceEffects(unit)
+      unit:RemoveModifierByName("modifier_item_lucience_movespeed_effect")
+    end
+    -- Force reapplication of effect modifiers so that effects update immediately when Lucience is upgraded
+    foreach(RemoveLucienceEffects, units)
+  end
+end
+
+function modifier_item_lucience_movespeed_aura:GetModifierAura()
+  return "modifier_item_lucience_movespeed_effect"
+end
+
+------------------------------------------------------------------------
+
+modifier_item_lucience_regen_effect = class({})
+
+function modifier_item_lucience_regen_effect:OnCreated()
+  self.regenBonus = self:GetAbility():GetSpecialValueFor("regen_bonus")
+end
+
+function modifier_item_lucience_regen_effect:DeclareFunctions()
+  return {
+    MODIFIER_PROPERTY_HEALTH_REGEN_CONSTANT
+  }
+end
+
+function modifier_item_lucience_regen_effect:GetModifierConstantHealthRegen()
+  return self.regenBonus
+end
+
+function modifier_item_lucience_regen_effect:GetEffectName()
+  return "particles/units/heroes/hero_necrolyte/necrolyte_ambient_glow.vpcf"
+end
+
+function modifier_item_lucience_regen_effect:GetTexture()
+  return regenIconName
+end
+
+------------------------------------------------------------------------
+
+modifier_item_lucience_movespeed_effect = class({})
+
+function modifier_item_lucience_movespeed_effect:OnCreated()
+  self.movespeedBonus = self:GetAbility():GetSpecialValueFor("speed_bonus")
```

Comment:
Well, the weirder part is that Lucience effects don't seem to update properly. Even now that I've put in `OnRefresh`. Could be related to the item not directly giving the aura modifier, but I don't know.